server/pulp/server/webservices/views/users.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+
+from django.views.generic import View
+from django.core.urlresolvers import reverse
+
+from pulp.server import exceptions as pulp_exceptions
+from pulp.server.auth import authorization
+from pulp.server.db.model.auth import Permission
+from pulp.server.managers import factory
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views.util import (generate_json_response,
+                                                generate_json_response_with_pulp_encoder,
+                                                generate_redirect_response,
+                                                json_body_required)
+
+
+USER_WHITELIST = [u'login', u'name', u'roles']
+
+
+class UsersView(View):
+    """
+    Views for users.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        List all users.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing a list of users
+        :rtype: django.http.HttpResponse
+        """
+        query_manager = factory.user_query_manager()
+        users = query_manager.find_all()
+
+        for user in users:
+            add_link(user)
+            process_dictionary_against_whitelist(user, USER_WHITELIST)
+
+        return generate_json_response_with_pulp_encoder(users)
+
+    @auth_required(authorization.CREATE)
+    @json_body_required
+    def post(self, request):
+        """
+        Create a new user.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing the user
+        :rtype: django.http.HttpResponse
+        """
+        # Pull all the user data
+        user_data = request.body_as_json
+        login = user_data.pop('login', None)
+        if login is None:
+            raise pulp_exceptions.MissingValue(['login'])
+        password = user_data.pop('password', None)
+        name = user_data.pop('name', None)
+        if user_data:
+            raise pulp_exceptions.InvalidValue(user_data.keys())
+        # Creation
+        manager = factory.user_manager()
+        args = [login]
+        kwargs = {'password': password,
+                  'name': name}
+
+        user = manager.create_user(*args, **kwargs)
+
+        # Add the link to the user
+        link = add_link(user)
+
+        # Grant permissions
+        permission_manager = factory.permission_manager()
+        permission_manager.grant_automatic_permissions_for_resource(link['_href'])
+
+        response = generate_json_response_with_pulp_encoder(user)
+        return generate_redirect_response(response, link['_href'])
+
+
+class UserView(View):
```

Comment:
Maybe change to UserResource? That has been the convention, and User vs Users is not visually obvious.